data/transactions/logic/evalStateful_test.go

```diff
@@ -592,20 +592,23 @@ pop
 !=
 `
 	type desc struct {
-		source string
-		eval   func([]byte, EvalParams) (bool, error)
-		check  func([]byte, EvalParams) error
+		source       string
+		eval         func([]byte, EvalParams) (bool, error)
+		evalStateful func([]byte, EvalParams) (int, bool, error)
+		check        func([]byte, EvalParams) error
```

Comment:
I think I'm going to argue for a shared pointer to a single cost budget counter.  Each ep would point to the same place.  It avoids leaking the cost into the call and returning interface and creating this difference.